### PR TITLE
search functionality

### PR DESCRIPTION
**Pull request by Isabirye1515**
Screenshot from 2024-06-09 02-23-29

Screenshot from 2024-06-09 02-09-05

[Screencast from 09-06-24 01:42:53.webm](https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/150329678/cb3fbd2a-9798-4954-b68e-9016a93052aa)

### CODE REVIEW

**Comment by mozzy11** [comment]
Thanks @Isabirye1515 .
Can you add a screen recording ??

**Comment by Isabirye1515** [comment]
[Screencast from 11-06-24 10:28:05.webm](https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/150329678/e2c98f22-681e-48ce-b67b-7c7c82c6faed)

**Comment by josephbate** [comment]
well done @Isabirye1515

**Comment by Isabirye1515** [comment]
thanks bro @josephbate and Mr @mozzy11

**Review by mozzy11 — CHANGES_REQUESTED** [review]
Thanks @Isabirye1515 for attemting to do this.
I like the search results bar you have implemeneted
This search button should only suport patient search

- by first name , 
 - last name, 
 - National Id
 - Patient ID
 - Unique Health Id
 - Date Of Birth. 
 
 The serach Results should be Autocomplete where by matching results are displayed as user is typing

File: frontend/src/components/layout/search/searchBar.js [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  };
+
+  const handleClearHistory = () => {
+    setSearchHistory([]);
+    localStorage.removeItem("searchHistory");
+  };
+
+  const handleSearch = () => {
+    const isAccessionNumber = /^\d+$/.test(searchInput.trim());
+    if (isAccessionNumber) {
```
Comment (mozzy11) [inline frontend/src/components/layout/search/searchBar.js:79]: fo the start, lets implement only Patient Search

File: frontend/src/components/Style.css [hunk 2]
```diff
@@ -536,3 +536,8 @@ button {
     width: 14rem;
   }
 }
+.patientHeader{
```
Comment (mozzy11) [inline frontend/src/components/Style.css:539]: you need to run `npm run format `to properly fromat your frontend code

**Comment by Isabirye1515** [comment]
Ok thank sir

**Comment by Isabirye1515** [comment]
[Screencast from 16-06-24 03:02:05.webm](https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/150329678/57bfe377-fcfc-4713-896a-32a1da7c980b)

**Pull request closed without merging**